cms/server/templates/contest/task_submissions.html

```diff
@@ -101,21 +101,41 @@ <h2 style="margin-bottom: 10px">{{ _("Submit a solution") }}</h2>
 {% end %}
 
 <div id="submit_solution" class="row">
-    <div class="span5">
+    <div class="span9">
         <form class="form-horizontal" enctype="multipart/form-data" action="{{ url_root }}/tasks/{{ encode_for_url(task.name) }}/submit" method="POST">
-            <fieldset>
+            <fieldset class="row-fluid">
 {% for idx, filename in enumerate(x.filename for x in task.submission_format) %}
                 <div class="control-group">
-                    <label class="control-label" for="input{{ idx }}">{{ filename.replace(".%l", "") }}</label>
-                    <div class="controls">
+                        <label class="control-label span2" for="input{{ idx }}">{{ filename.replace(".%l", "") }}</label>
+
+                    <div class="span4">
                         <input type="file" class="input-xlarge" id="input{{ idx }}" name="{{ filename }}"/>
+                        <!-- FIXME: figure out why we had div class="controls" wrapping this -->
+                    </div>
+
+                    <!-- FIXME: What does this even do when there are multiple inputs? -->
```

Comment:
Oh, wait, no, my mistake; this likely causes CMS to be Very Angry because now it's not autodetecting the language type, it's using whatever the user selected from the dropdown (so will always see them as the same type thanks to `which_language`. In which case it will 100% compiler error because it'll try to compile e.g. `a.cpp b.cpp c.java` all with `g++`